app/services/cards.ts

```diff
@@ -0,0 +1,259 @@
+}
+
+export enum Faction {
+  Alliance = 'Alliance',
+  Horde = 'Horde',
+  Neutral = 'Neutral'
+}
+
+export interface Mechanic {
+  name: Name;
+}
+
+export enum Name {
+  AIMustPlay = 'AIMustPlay',
+  Adapt = 'Adapt',
+  AdjacentBuff = 'AdjacentBuff',
+  AffectedBySpellPower = 'AffectedBySpellPower',
+  Aura = 'Aura',
+  AvengeX = 'Avenge (X)',
+  Battlecry = 'Battlecry',
+  BloodGem = 'Blood Gem',
+  CastsWhenDrawn = 'Casts When Drawn',
+  Charge = 'Charge',
+  Combo = 'Combo',
+  Corrupt = 'Corrupt',
+  Deathrattle = 'Deathrattle',
+  Discover = 'Discover',
+  DivineShield = 'Divine Shield',
+  Echo = 'Echo',
+  Freeze = 'Freeze',
+  Frenzy = 'Frenzy',
+  ImmuneToSpellpower = 'ImmuneToSpellpower',
+  Inspire = 'Inspire',
+  InvisibleDeathrattle = 'InvisibleDeathrattle',
+  Invoke = 'Invoke',
+  JadeGolem = 'Jade Golem',
+  Lifesteal = 'Lifesteal',
+  Magnetic = 'Magnetic',
+  Morph = 'Morph',
+  NatureSpellDamage = 'Nature Spell Damage',
+  OneTurnEffect = 'OneTurnEffect',
+  Outcast = 'Outcast',
+  Overkill = 'Overkill',
+  Overload = 'Overload',
+  Poisonous = 'Poisonous',
+  Quest = 'Quest',
+  Questline = 'Questline',
+  Reborn = 'Reborn',
+  Recruit = 'Recruit',
+  Rush = 'Rush',
+  Secret = 'Secret',
+  Sidequest = 'Sidequest',
+  Silence = 'Silence',
+  SpellDamage = 'Spell Damage',
+  Spellburst = 'Spellburst',
+  Stealth = 'Stealth',
+  Summoned = 'Summoned',
+  Taunt = 'Taunt',
+  Tradeable = 'Tradeable',
+  Twinspell = 'Twinspell',
+  Windfury = 'Windfury'
+}
+
+export enum Race {
+  All = 'All',
+  Beast = 'Beast',
+  Demon = 'Demon',
+  Dragon = 'Dragon',
+  Elemental = 'Elemental',
+  Mech = 'Mech',
+  Murloc = 'Murloc',
+  Pirate = 'Pirate',
+  Quilboar = 'Quilboar',
+  Totem = 'Totem'
+}
+
+export enum Rarity {
+  Common = 'Common',
+  Epic = 'Epic',
+  Free = 'Free',
+  Legendary = 'Legendary',
+  Rare = 'Rare'
+}
+
+export enum SpellSchool {
+  Arcane = 'Arcane',
+  Fel = 'Fel',
+  Fire = 'Fire',
+  Frost = 'Frost',
+  Holy = 'Holy',
+  Nature = 'Nature',
+  Shadow = 'Shadow'
+}
+
+export enum Type {
+  Enchantment = 'Enchantment',
+  Hero = 'Hero',
+  HeroPower = 'Hero Power',
+  Minion = 'Minion',
+  Spell = 'Spell',
+  Weapon = 'Weapon'
+}
+
+export interface CardsInfo {
+  patch: string;
+  classes: string[];
+  sets: string[];
+  standard: string[];
+  wild: string[];
+  types: string[];
+  factions: string[];
+  qualities: string[];
+  races: string[];
+  locales: Locales;
+}
+
+export interface Locales {
+  DE_DE: string;
+  EN_GB: string;
+  EN_US: string;
+  ES_ES: string;
+  ES_MX: string;
+  FR_FR: string;
+  IT_IT: string;
+  KO_KR: string;
+  PL_PL: string;
+  PT_BR: string;
+  RU_RU: string;
+  ZH_CN: string;
+  ZH_TW: string;
+  JA_JP: string;
+  TH_TH: string;
+}
+
+export async function getAllCards(): Promise<Card[]> {
+  try {
+    const response = await card.get<Card[]>('cards');
+    return response.data;
+  } catch (error) {
+    throw externalApiError(`External Api Error ${error}`);
```

Comment:
it is good practice to describe where the error was, in this case it is unclear if it was this method or the method below.
throw externalApiError(External api error getting all cards ${error});

Reply:
add describe error for two methods